### PR TITLE
[webgpu] Add transpose shared kernel

### PR DESCRIPTION
**Pull request by axinging**
FEATURE
For transpose [1024, 1024], shared memory got a ~15% improvement.
For transpose [1500, 2048], shared memory got a ~40% improvement.
 
Fix: https://github.com/tensorflow/tfjs/issues/1625

Naive ([1024, 1024]), trials = 10, rep = 10);
Mean time: 17.813 ms -> 1.781 / rep
Mean time: 18.380 ms -> 1.838 / rep
Mean time: 18.434 ms -> 1.843 / rep

Shared ([1024, 1024]), trials = 10, rep = 10);
Mean time: 14.341 ms -> 1.434 / rep
Mean time: 14.187 ms -> 1.419 / rep
Mean time: 14.737 ms -> 1.474 / rep

Naive ([1500, 2048]), trials = 10, rep = 10);
Mean time: 67.197 ms -> 6.720 / rep
Mean time: 68.796 ms -> 6.880 / rep
Mean time: 70.494 ms -> 7.049 / rep

Shared ([1500, 2048]), trials = 10, rep = 10);
Mean time: 37.334 ms -> 3.733 / rep
Mean time: 37.655 ms -> 3.765 / rep
Mean time: 37.517 ms -> 3.752 / rep

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2447)

### CODE REVIEW

**Comment by math3d** [comment]
@annxingyuan , PTAL

**Comment by axinging** [comment]
@annxingyuan PTAL

File: tfjs-backend-webgpu/src/backend_webgpu.ts [hunk 1]
```diff
@@ -851,6 +852,10 @@ export class WebGPUBackend extends KernelBackend {
     if (this.shouldExecuteOnCPU([x])) {
       return this.cpuBackend.transpose(x, perm);
     }
+    if (x.shape.length === 2 && perm[0] === 1 && perm[1] === 0) {
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/backend_webgpu.ts]: Replace this if statement with `util.arraysEqual(perm, [1, 0])`.

**Comment by axinging** [comment]
Done, PTAL! 
Thanks @annxingyuan!

**Comment by annxingyuan** [comment]
Thank you, @axinging !

**Pull request merged**